kibble/api/languages.go

```diff
@@ -64,6 +70,45 @@ func loadAllLanguagesFromApi(cfg *models.Config, site *models.Site) error {
 	return nil
 }
 
+// Returns an error if:
+// - default language is not set.
+// - default language is missing display name.
+// - no site languages are set.
+// - any site language is missing a display name.
+// - default language is not included in supported languages.
+func (l languagesV1) validate() error {
+	isZero := func(val string) bool {
+		return val == ""
+	}
+
+	if isZero(l.DefaultLanguage.Code) && isZero(l.DefaultLanguage.Label) && isZero(l.DefaultLanguage.Name) {
+		return errors.New("DefaultLanguage not set")
+	} else if isZero(l.DefaultLanguage.Name) {
+		return errors.New(fmt.Sprintf("DefaultLanguage %s is missing display name", l.DefaultLanguage.Code))
+	}
+
+	if len(l.SupportedLanguages) == 0 {
+		return errors.New("No SiteLanguages set")
+	}
+
+	defaultInLanguages := false
+	for _, lang := range l.SupportedLanguages {
+		if lang.Code == l.DefaultLanguage.Code {
+			defaultInLanguages = true
+		}
+
+		if isZero(lang.Name) {
+			return errors.New(fmt.Sprintf("Language %s is missing display name", lang.Code))
+			// no need to check Code or Label as they are required by database schema.
```

Comment:
The placement of this comment is weirding me out. I've got two conflicting thoughts: It seems like reasonable info to readers but I'd rather not have it at all.

---

Not a suggestion for this PR, but is there a reason why the database doesn't enforce this also? It would mean only having to validate the `defaultInLanguages` with an early `break`.